File: 3612-adjacent-increasing-subarrays-detection-i/3612-adjacent-increasing-subarrays-detection-i.cpp
class Solution {
public:
    bool hasIncreasingSubarrays(vector<int>& nums, int k) {
        if (nums.size() < 2) 
            return false;
        if (nums.size() >= 2 && k == 1) 
            return true;

        int pl = 0;
        int l = 1;

        for (int i = 1; i < nums.size(); ++i) {
            if (nums[i] > nums[i - 1]) {
                l += 1;
                if (pl >= k && l >= k)
                    return true;
                if (l == 2 * k)
                    return true;
            } else {
                pl = l;
                l = 1;
            }
        }
        return false;

    }
};

File: 3612-adjacent-increasing-subarrays-detection-i/3612-adjacent-increasing-subarrays-detection-i.py
class Solution:
    def hasIncreasingSubarrays(self, nums: List[int], k: int) -> bool:
        if len(nums) < 2: 
            return False
        if len(nums) >= 2 and k == 1: 
            return True
        pl = 0
        l = 1
        for i in range(1,len(nums)):
            if nums[i] > nums[i-1]:
                l += 1
                if pl >= k and l >= k:
                    return True
                if l == 2*k:
                    return True
            else:
                pl = l
                l = 1
        return False



        